### ISSUE TITLE
Catalog updates should not use hard coded value for /configs directory during catalog rebuilds

### ISSUE THREAD

**Issue by jchunkins** [openshift/oc-mirror#417]
# Version

N/A ... evaluated the master branch reviewing the code

# What happened?

I was evaluating the code to see how the operator catalogs are built within oc-mirror and noticed that the code which processes an existing catalog using this process:
1. delete the `/configs` directory using a "whiteout file"
2. add back in the current contents of the FBC into the `/configs` directory

This process takes place in the `processCatalogRefs` function [here](https://github.com/openshift/oc-mirror/blob/1f3f72569ac31ee4bacde46f156a31618b752b9e/pkg/cli/mirror/catalog_images.go#L127).

The problem as I see it, is that the `/config` directory is hard coded and does not attempt to take into account the possibility of re-defining the location of where your FBC content resides via the `operators.operatorframework.io.index.configs.v1` label. So if I create my catalog image using file path called `/destination` like this example:

```Containerfile
# The base image is expected to contain
# /bin/opm (with a serve subcommand) and /bin/grpc_health_probe
FROM quay.io/operator-framework/opm:latest

# Configure the entrypoint and command
ENTRYPOINT ["/bin/opm"]
CMD ["serve", "/destination"]

# Copy declarative config root into image at /configs
ADD mysource /destination

# Set DC-specific label for the location of the DC root directory
# in the image
LABEL operators.operatorframework.io.index.configs.v1=/destination
```

I believe that the code as it is written would attempt to blindly update the hard coded path `/configs` rather than the path defined by my catalog.

# What did you expect to happen?

I would expect that the code would honor the source catalog's `operators.operatorframework.io.index.configs.v1` directory

# How to reproduce it (as minimally and precisely as possible)?

Create a mirror-config.yaml that has an operator entry in it

```console
$ oc-mirror --config mirror-config.yaml docker://localhost:5000
```

# Anything else we need to know?

Note I did NOT run the oc-mirror to validate any behavior... my report is based solely on reviewing the code. Happy to be proven wrong about what the code is actually doing.

# References

None

**Comment by jpower432** [comment]
Hi @jchunkins, thanks for the question/issue. You are correct. `oc-mirror` does honor this location when mirroring the requested catalog and processing the FBC contents, but when rebuilding the catalogs it does use the `/configs` directory as the place to store FBC content. The `CMD` is updated as well as the `operators.operatorframework.io.index.configs.v1` label. I don't see any issue with honoring the current location instead of using `/configs`.

**Comment by jchunkins** [comment]
@jpower432 So if I understand correctly the process is:

1. Read original catalog 
2. Process original catalog honoring the `operators.operatorframework.io.index.configs.v1` label
3. Create replacement catalog based on the original catalog, but now it uses new FBC contents with updated `CMD` and  `operators.operatorframework.io.index.configs.v1` to use `/configs` instead.

Do I have that high level process right?

Since the new catalog contents will now reside in `/configs` in the resulting catalog, the old directory might still be present, but effectively it won't be used. Is that right?

**Comment by jpower432** [comment]
@jchunkins. Yep, you have it right. I do think we could support custom catalogs better in this method by preserving their labels instead of overwriting them, but if the original concern was that the original FBC would not be processed correctly, we should be good there.

**Comment by jchunkins** [comment]
@jpower432 Thanks for the clarification on all of this. I did not have the full picture, and I interpreted the hard coded values as a sign that the label defined in the original catalog might not be honored (i.e. I thought the code was making assumptions that it was always `/configs`). 

It still seems like the process as described potentially leaves the original folder (i.e. `/destination` in my example) in place because the "whiteout file layer" (i.e. `/.wh.configs`) as defined only blows away the `/configs` directory and replaces it with the new FBC content. In other words, the replacement catalog is based on the original catalog, and the original catalog still has `/destination` since it is never cleaned out with its own "whiteout file layer". Not the end of the world kind of problem but could be confusing for someone poking around inside a catalog image and finding two different directories... one that's used, and another that's just left over stuff.
